### PR TITLE
Add an equivalent definition of countability to iset.mm 

### PR DESCRIPTION
**Pull request by jkingdon**
This pull request proves:

```
    $( A set is countable iff there is a surjection from a decidable subset of
       the natural numbers onto it.  Based on a definition proposed by David
       theHigherGeometer oneline.  The decidability condition is needed as
       shown at ~ ctssdc .  (Contributed by Jim Kingdon, 15-Aug-2023.) $)
    ctssdc $p |- ( E. s ( s C_ _om /\ E. f f : s -onto-> A
        /\ A. n e. _om DECID n e. s )
        <-> E. f f : _om -onto-> ( A |_| 1o ) ) $=
```

The motivation is a discussion thread starting with https://mathstodon.xyz/@highergeometer/110892635391287895 especially the part about "I probably should have done the even better definition which is that there is a surjection from a [decidable] subset of NN, which means you don't need to worry about the empty set, and also the finite case is more cleanly captured"

### CODE REVIEW

**Comment by jkingdon** [comment]
I updated the comment based on https://mathstodon.xyz/@highergeometer/110913478065960269 . There probably isn't much value to a vague citation like that anyway (and a specific URL is likely to bit-rot at least on a scale of years).

**Review by benjub — APPROVED** [review]

**Comment by benjub** [comment]
> I updated the comment based on https://mathstodon.xyz/@highergeometer/110913478065960269 . There probably isn't much value to a vague citation like that anyway (and a specific URL is likely to bit-rot at least on a scale of years).

That's an interesting discussion. Maybe you can find a way to write the gist of it somewhere in a theorem comment or section comment in iset.mm or in index.html ? (or maybe it's already there, I haven't checked).

Having these results formalized in metamath is very nice.

From the picture, I guess that you are "soaproot" there. Do you know who "Higher Geometer" is ? And more importantly, do you know who ℵ_ω / ax_groth is ? I think it is problematic that that user uses the Metamath logo. Since his description only says "check out https://us.metamath.org/", one could think this is the official account of Metamath.

**Comment by DavidMichaelRoberts** [comment]
@benjub I'm HigherGeometer :-)

File: iset.mm [hunk 1]
```diff
@@ -67983,6 +67983,135 @@ property of disjoint unions (see ~ updjud in the case of functions).
...
+      N $.
+  $}
+
+  ${
+    $d A f g m s $.  $d A f g x $.  $d A f m s y z $.  $d f m n s y z $.
+    $d A n $.
+    $( A set is countable iff there is a surjection from a decidable subset of
+       the natural numbers onto it.  The decidability condition is needed as
+       shown at ~ ctssdc .  (Contributed by Jim Kingdon, 15-Aug-2023.) $)
+    ctssdc $p |- ( E. s ( s C_ _om /\ E. f f : s -onto-> A
```
Comment (icecream17) [inline iset.mm:68082]: the theorem is quoting itself? + maybe you meant ctssexmid below

(approve when done)

**Comment by jkingdon** [comment]
> That's an interesting discussion. Maybe you can find a way to write the gist of it somewhere in a theorem comment or section comment in iset.mm or in index.html ? (or maybe it's already there, I haven't checked).

Well, the current pull request is for some of it and https://github.com/metamath/set.mm/issues/3396 and https://github.com/metamath/set.mm/issues/3394 are for others. As a formal math project, I'd have us focus on what we prove rather than trying to recreate wikipedia or ncatlab.

> From the picture, I guess that you are "soaproot" there.

Correct. 

> And more importantly, do you know who ℵ_ω / ax_groth is ? I think it is problematic that that user uses the Metamath logo. Since his description only says "check out https://us.metamath.org/", one could think this is the official account of Metamath.

Based on a quick browse, appears to be a student who is a fan of metamath. I guess to me it doesn't look much like an official account, so I'm not especially concerned, but whatever happens I hope we try to encourage their enthusiasm. Who knows, maybe they'll end up being more involved with the project at some point.

File: iset.mm (hunk 1 above)
Comment (jkingdon) [reply iset.mm]: Yes, it is supposed to be ctssexmid

Thanks for noticing this.

**Pull request merged**